app/models/custom_value.rb

```diff
@@ -79,16 +97,9 @@ def validate_format_of_value
 
   def validate_type_of_value
     if value.present?
-      # Format specific validations
-      case custom_field.field_format
-      when 'int'
-        errors.add(:value, :not_a_number) unless value =~ /\A[+-]?\d+\z/
-      when 'float'
-        begin; Kernel.Float(value); rescue; errors.add(:value, :invalid) end
-      when 'date'
-        errors.add(:value, :not_a_date) unless value =~ /\A\d{4}-\d{2}-\d{2}\z/
-      when 'list'
-        errors.add(:value, :inclusion) unless custom_field.possible_values.include?(value)
+      result = strategy.validate_type_of_value
+      if result
+        errors.add(:value, result)
```

Comment:
This is ether a question of naming or of correctness. 

If it is just a question of naming you should rename this variable 'error' instead of 'result' because you are expecting a single error to be returned or nothing at all. 

If it is a question of correctness, then I'd say the implementation is wrong :). You do not know how many errors might be returned. You could either move the whole error setting to within the strategy. I think that is ok because the strategy is a part of the custom value.  Or you accept multiple error to be returned. IMO, assigning the errors within the strategy is preferable.